packages/data/src/onboarding/selectors.ts

```diff
@@ -0,0 +1,111 @@
+import { WPDataSelectors } from '../types';
+
+export const getProfileItems = (
+	state: OnboardingState
+): ProfileItemsState | Record< string, never > => {
+	return state.profileItems || {};
+};
+
+export const getTasksStatus = (
+	state: OnboardingState
+): TasksStatusState | Record< string, never > => {
+	return state.tasksStatus || {};
+};
+
+export const getOnboardingError = (
+	state: OnboardingState,
+	selector: string
+): unknown | false => {
+	return state.errors[ selector ] || false;
+};
+
+export const isOnboardingRequesting = (
+	state: OnboardingState,
+	selector: string
+): boolean => {
+	return state.requesting[ selector ] || false;
+};
+
+// Types
+export type OnboardingSelectors = {
+	getProfileItems: () => ReturnType< typeof getProfileItems >;
+	getTasksStatus: () => ReturnType< typeof getTasksStatus >;
+	getOnboardingError: () => ReturnType< typeof getOnboardingError >;
+	isOnboardingRequesting: () => ReturnType< typeof isOnboardingRequesting >;
+} & WPDataSelectors;
+
+export type OnboardingState = {
+	profileItems: ProfileItemsState;
+	tasksStatus: TasksStatusState;
+	// TODO clarify what the error record's type is
+	errors: Record< string, unknown >;
+	requesting: Record< string, boolean >;
+};
+
+export type TasksStatusState = {
+	automatedTaxSupportedCountries: string[];
+	enabledPaymentGateways: string[];
+	hasHomepage: boolean;
+	hasPaymentGateway: boolean;
+	hasPhysicalProducts: boolean;
+	hasProducts: boolean;
+	isAppearanceComplete: boolean;
+	isTaxComplete: boolean;
+	shippingZonesCount: number;
+	stripeSupportedCountries: string[];
+	stylesheet: string;
+	taxJarActivated: boolean;
+	// TODO - fill out this type
+	themeMods: unknown;
+	wcPayIsConnected: false;
+};
+
+export type Industry = {
+	slug: string;
+};
+
+export type ProductCount = '0' | '1-10' | '11-100' | '101 - 1000' | '1000+';
```

Comment:
I've used the specific available string values here because I think it's self documenting and helpful to the next person trying to figure out what onboarding does.